---
sort: 2
---

# -Omics

source: `{{ page.path }}`

{% include list.liquid all=true %}

<span class="badge badge-info">Metagenomic tools</span>

## Metagenomics

:link: **Useful resources**

* [RNAseq analysis](https://bioinformatics-core-shared-training.github.io/RNAseq-R/) - a link to similar content of RNAseq by combine Australia is also available [here](http://combine-australia.github.io/RNAseq-R/)
* [Next Gen Seek](http://nextgenseek.com)
* [Bits of DNA](http://liorpachter.wordpress.com/seq)
* [RNA-Seq Blog](http://www.rna-seqblog.com)
* [Journal of Next Generation Sequencing & Applications](http://www.omicsonline.org/next-generationsequencing-
applications.php)
* [CoreGenomics](http://core-genomics.blogspot.com)
* [Next-Gen Sequencing](http://nextgenseq.blogspot.com)
* [Omics! Omics!](http://omicsomics.blogspot.com)
* [In Between Lines of Code](http://flxlexblog.wordpress.com)
* [Kevinâ€™s GATTACA World](http://kevin-gattaca.blogspot.com)
* [Blog @ Illumina](http://blog.illumina.com)
* [Next Generation Technologist](http://www.yuzuki.org)
* [Applied compuational genomics](https://github.com/quinlan-lab/applied-computational-genomics)
* [OSSU bioinformatics](https://github.com/ossu/bioinformatics)
* [AusMicrobiome metagenome](https://github.com/martinostrowski/metagenome) - *In development* An assembly and annotation pipeline using standardised tools to characterise shotgun metagenomes produced by the Marine Microbes and Australian Microbiome Project.

Detailed tutorials

- [Computational Genomics Tutorial](https://genomics.sschmeier.com/index.html) - This is an introductory tutorial for learning computational genomics mostly on the Linux command-line. You will learn how to analyse next-generation sequencing (NGS) data. The data you will be using is real research data. The final aim is to identify genome variations in evolved lines of bacteria that can explain the observed biological phenotypes. 

Inspiration for some RShiny apps by [imetalab](https://shiny.imetalab.ca/)

# Incorporating related -Omics data

- [MICOM](https://micom-dev.github.io/micom/) package for metabolic modeling of microbial communities. MICOM allows you to construct a community model from a list on input COBRA models and manages exchange fluxes between individuals and individuals with the environment. It explicitly accounts for different abundances of individuals in the community and can thus incorporate data from 16S amplicon sequencing or metagenomic experiments. It allows optimization with a variety of algorithms modeling the trade-off between egoistic growth rate maximization and cooperative objectives
- [Computational Methods and Data Analysis for Metabolomics Supplemental site for the Springer Nature book](https://metabolomics-data.github.io/). This includes a variety of links and examples for analysis of metabolomics data
- [iMetaShiny apps](https://shiny.imetalab.ca/) - Metalab represents the whole framework, including the Metalab Desktop version, cloud implementation, automatic MetaReport, shiny apps for data visualization and analysis. We aim to make iMetaLab platform a free and one-stop toolset for metaproteomics, with increasing amount of tools under active development.
- [Huttenhower lab](https://huttenhower.sph.harvard.edu/tools/) - links to a range of pipelines and tools related to microbial community analysis including a range of -omics data.
